art/attacks/boundary.py

```diff
@@ -69,6 +70,12 @@ def __init__(self, classifier, targeted=True, delta=0.01, epsilon=0.01, step_ada
         :type init_size: `int`
         """
         super(BoundaryAttack, self).__init__(classifier=classifier)
+        if not isinstance(classifier, ClassifierNeuralNetwork) or not isinstance(classifier, ClassifierGradients):
```

Comment:
Yes, `BoundaryAttack` only requires `predict` and should only require `ClassifierGradients`. Currently `BoundaryAttack` seems to be implicitly implemented for image data in the lines following https://github.com/IBM/adversarial-robustness-toolbox/blob/dce8d74de74bc6560c0a0fbac3ee1d5d451431c6/art/attacks/boundary.py#L279 

I have a fix to make it work for 2-dimensional data, but I would like to dicuss it first with @minhitbk who implemented these lines.

Reply:
Ah, ok. We could try to resolve it with this PR since it introduced the split in interfaces, or open an issue. Having it resolved together might be preferable.

Reply:
Ok, I'll do some more testing and update BoundaryAttack soon.

Reply:
Resolved in df504d916235e97ed68d204afc7cc435bce2168a

Reply:
Isn’t it still assuming the model is an instance of `ClassifierGradients` now?

Reply:
You are right, I didn't notice it because I tested it with `LogisticRegression` and `SVC`, which both have gradients. Resolved with 6231b2b6fe08c4373d877183827b1e52d5a857a4